### ISSUE TITLE
Unable to use union as an argument of a directive

### ISSUE THREAD

**Issue by MartinPavlik** [dotansimha/graphql-code-generator#1462]
**Describe the bug**
Hi, I'm trying to write my custom plugin and I'm having problem with using 

Schema:
```graphql
  union Any = String | Int | Float | ID

  directive @default(
    value: Any,
  ) on ENUM_VALUE | FIELD_DEFINITION

  type CardEdge {
    count: Int! @default(value: 1)
  }
```

codegen.yml
```yml
schema: input-schema.graphql
overwrite: true
generates:
  ./output.ts:
      - plugin.js
  ./types.d.ts:
    plugins:
      - typescript-common
      - typescript-server
  schema.graphql:
    plugins:
      - schema-ast
```

Then I run the schema thru graphql-codegen and it crashes with this error:
![error](https://user-images.githubusercontent.com/7362265/54370582-227abe00-4678-11e9-8de3-50d8d30cbcea.png)

It's weird that the plugin `schema-ast` has no problem with it, but `typescript-common` and/or `typescript-server` do. In my plugin I use only `schemaToTemplateContext`. I tried to search codebase of this project for `Unkown type` but without any luck. Any idea what causes this error?

Thanks in advance :sweat_smile: 

**Expected behavior**
No error

**Environment:**
 - OS: Ubuntu 16
 - Codegen: 0.18.0
 - Node: 10.15.3

**Comment by dotansimha** [comment]
Hi @MartinPavlik 
The `Unkown type` error comes directly from `graphql`. Any chance you can create a repo with a reproduction? 
Also, we welcome you to use the latest alpha version, it has better way to implement plugins than `schemaToTemplateContext` (using AST visitor).
It's still in alpha, but the updates docs are available here: https://5c8bd8a4a550500008ff7d95--graphql-code-generator.netlify.com/docs/custom-codegen/using-visitor

**Comment by dotansimha** [comment]
Added test and fixed it: https://github.com/dotansimha/graphql-code-generator/pull/1483

**Comment by MartinPavlik** [comment]
@dotansimha Thanks

**Comment by dotansimha** [comment]
Fix available in 1.0.0 👍
